python/pyarrow/scalar.pxi

```diff
@@ -240,6 +240,10 @@ cdef class UInt8Scalar(Scalar):
         cdef CUInt8Scalar* sp = <CUInt8Scalar*> self.wrapped.get()
         return sp.value if sp.is_valid else None
 
+    def __int__(self):
+
+        return(self.as_py())
+
```

Comment:
Is all this copying and pasting OK?  I could define a new scalar integer subclass which defines this once and then have the existing int-based subclasses be subclasses of that, but I wasn't sure if that would be overengineering things?

Reply:
I would be nice to have a Scalar subclass from which all existing int-based classes would inherit from, like in [BaseExtensionType(DataType)](https://github.com/apache/arrow/blob/40c5300edc6ccc0ea4a9131722628ac6203ce806/python/pyarrow/types.pxi#L1644). _But_ we would need to do some gymnastic to move `as_py` to this newly created, lets say `BaseIntScalar` class. That, I think, would make it more of an effort in the end.